contracts/script/setup.ts

```diff
@@ -1,219 +1,349 @@
   type Client,
   type GetContractReturnType,
-  type Transport,
 } from "viem";
-
-import type { artifacts } from "@rocketh";
+import { waitForTransactionReceipt } from "viem/actions";
 import { mnemonicToAccount } from "viem/accounts";
-import { readConfig } from "./readConfig.js";
-
-const l1Config = {
-  port: 8545,
-  chainId: 31337,
-};
-
-const l2Config = {
-  port: 8546,
-  chainId: 31338,
-};
-
-function createDeploymentGetter<
-  transport extends Transport = Transport,
-  chain extends Chain | undefined = Chain | undefined,
-  account extends Account | undefined = Account | undefined,
-  client extends Client<transport, chain, account> = Client<
-    transport,
-    chain,
-    account
-  >,
->(
-  environment: Environment<
-    UnresolvedUnknownNamedAccounts,
-    UnresolvedNetworkSpecificData,
-    UnknownDeployments
-  >,
-  client: client,
+import { artifacts } from "@rocketh";
+import { rmdir } from "node:fs/promises";
+
+import { deployArtifact } from "../test/fixtures/deployArtifact.js";
+import { urgArtifact } from "../test/fixtures/externalArtifacts.js";
+import { UncheckedRollup } from "../lib/unruggable-gateways/src/UncheckedRollup.js";
+import { WebSocketProvider } from "ethers/providers";
+import { Gateway } from "../lib/unruggable-gateways/src/gateway.js";
+import { serve } from "@namestone/ezccip/serve";
+
+function createDeploymentGetter<C extends Client>(
+  environment: Environment,
+  client: C,
 ) {
-  return <TAbi extends Abi>(
-    name: string,
-  ): GetContractReturnType<TAbi, client> => {
+  return <ContractName extends keyof typeof artifacts>(
+    name: ContractName | string,
+  ) => {
     const deployment = environment.get(name);
     return getContract({
       abi: deployment.abi,
       address: deployment.address,
       client,
-    }) as unknown as GetContractReturnType<TAbi, client>;
+    }) as unknown as GetContractReturnType<
+      (typeof artifacts)[ContractName]["abi"],
+      C
+    >;
   };
 }
 
-/**
- * Sets up the cross-chain testing environment using blocksmith.js
- * @ref https://github.com/adraffy/blocksmith.js
- * @returns Environment with L1 and L2 contracts and a relayer
- */
-export async function setupCrossChainEnvironment() {
-  console.log("Setting up cross-chain ENS v2 environment...");
-
-  // Launch two separate Anvil instances for L1 and L2
-  const l1 = anvil(l1Config);
-  const l2 = anvil(l2Config);
-
-  await l1.start();
-  await l2.start();
-
-  console.log(`L1: Chain ID ${l1Config.chainId}, URL: ${l1.host}:${l1.port}`);
-  console.log(`L2: Chain ID ${l2Config.chainId}, URL: ${l2.host}:${l2.port}`);
-
-  // Deploy contracts to both chains
-  console.log("Deploying contracts...");
-
-  const l1Deploy = await executeDeployScripts(
-    resolveConfig(
-      await readConfig({
-        askBeforeProceeding: false,
-        network: "l1-local",
-      }),
-    ),
-  );
+export async function setupCrossChainEnvironment({
+  l2ChainId = 0xeeeeee,
+  l1ChainId = l2ChainId - 1,
```

Comment:
using 31337 or 1337 is probably better for consistency with pre-existing localhost detection